src/util/arith_tools.cpp

```diff
@@ -342,14 +395,50 @@ irep_idt bvrep_bitwise_op(
 }
 
 /// convert an integer to bit-vector representation with given width
+/// This uses two's complement for negative numbers.
+/// If the value is out of range, it is 'wrapped around'.
 irep_idt integer2bvrep(const mp_integer &src, std::size_t width)
 {
-  return integer2binary(src, width);
+  const mp_integer p = power(2, width);
+
+  if(src.is_negative())
+  {
+    // do two's complement encoding of negative numbers
+    mp_integer tmp = src;
+    tmp.negate();
+    tmp %= p;
+    if(tmp != 0)
+      tmp = p - tmp;
+    return integer2string(tmp, 16);
```

Comment:
This is not correct for negative numbers below `-power(2, width - 1)` (just consider `src = -255`, `width = 8`, which would result in `tmp = 1`).

Reply:
Well, but '1' is the right answer!

Reply:
Err, sorry, of course...